pkg/services/ngalert/store/alert_rule.go

```diff
@@ -566,22 +537,44 @@ func (st DBstore) GetAlertRulesForScheduling(ctx context.Context, query *ngmodel
 		query.ResultRules = rules
 
 		if query.PopulateFolders {
-			foldersSql := sess.Table("folder").Alias("d").Select("d.org_id, d.uid, d.title, d.parent_uid").
-				Where(`EXISTS (SELECT 1 FROM alert_rule a WHERE d.uid = a.namespace_uid AND d.org_id = a.org_id)`)
-			if len(disabledOrgs) > 0 {
-				foldersSql.NotIn("org_id", disabledOrgs)
+			query.ResultFoldersTitles = map[ngmodels.FolderKey]string{}
+			uids := map[int64]map[string]struct{}{}
+			for _, r := range rules {
+				om, ok := uids[r.OrgID]
+				if !ok {
+					om = make(map[string]struct{})
+					uids[r.OrgID] = om
+				}
+				om[r.NamespaceUID] = struct{}{}
 			}
+			for orgID, uids := range uids {
+				fakeUser := &user.SignedInUser{
+					UserID:           -1,
+					IsServiceAccount: true,
+					Login:            "grafana_scheduler",
+					OrgID:            orgID,
+					OrgRole:          org.RoleAdmin,
+					Permissions: map[int64]map[string][]string{
+						orgID: {
+							dashboards.ActionFoldersRead: []string{
+								datasources.ScopeAll,
```

Comment:
nit, for consistency:
```suggestion
								dashboards.ScopeFoldersAll,
```